### PR TITLE
feat(i18n): translate/localize references, cross-dataset references

### PR DESCRIPTION
**Pull request by rexxars**

### CODE REVIEW

**Review by jtpetty — APPROVED** [review]
LGTM - thanks for extracting those common blocks out

File: packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/ReferenceAutocomplete.tsx [hunk 1]
```diff
@@ -57,7 +59,12 @@ export const ReferenceAutocomplete = forwardRef(function ReferenceAutocomplete(
               <Box padding={4}>
                 <Flex align="center" height="fill" justify="center">
                   <StyledText align="center" muted>
-                    No results for <strong>“{props.searchString}”</strong>
+                    <Translate
+                      t={t}
+                      i18nKey="inputs.reference.no-results-for-query"
+                      components={{SearchTerm: ({children}) => <strong>{children}</strong>}}
```
Comment (ricokahler) [inline packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/ReferenceAutocomplete.tsx:65]: should this be removed now that simple tags are allowed in the template string?

File: packages/sanity/src/core/form/inputs/ReferenceInput/ReferenceStrengthMismatchAlertStrip.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * strong, or wise versa.
+ *
+ * @internal
+ */
+export function ReferenceStrengthMismatchAlertStrip({
+  actualStrength,
+  handleFixStrengthMismatch,
+}: {
+  actualStrength: 'weak' | 'strong'
+  handleFixStrengthMismatch: () => void
```
Comment (ricokahler) [inline packages/sanity/src/core/form/inputs/ReferenceInput/ReferenceStrengthMismatchAlertStrip.tsx:17]: not a blocking comment but should this be `onFixStrengthMismatch`?

**Review by ricokahler — APPROVED** [review]
nice clean up too! i like how you name keys

File: packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/ReferenceAutocomplete.tsx (hunk 1 above)
Comment (rexxars) [reply packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/ReferenceAutocomplete.tsx:65]: Ah thanks - copied from previous work, can simplify this now.

File: packages/sanity/src/core/form/inputs/ReferenceInput/ReferenceStrengthMismatchAlertStrip.tsx (hunk 2 above)
Comment (rexxars) [reply packages/sanity/src/core/form/inputs/ReferenceInput/ReferenceStrengthMismatchAlertStrip.tsx:17]: Lint rules says otherwise actually

**Pull request merged**